### ISSUE TITLE
Use ActionDispatch::Reloader instead of Shotgun

### ISSUE THREAD

**Issue by rzane** [tongueroo/jets#240]
## Summary

In development, even the simplest requests take multiple seconds. It's not because Jets is slow, but because `shotgun` is slow.

## Motivation

Rails is significantly faster than Jets in development because of this, despite the fact that Jets is significantly lighter weight than Rails.

## Reference-level explanation

Because Jets already relies on ActiveSupport::Dependencies, I think this should be possible?

The following middleware actually works for the simple example that I tried. It's definitely not thread-safe and there are probably a million issues, though.

```ruby
# config/environments/development.rb
class Reloader
  def initialize(app)
    @app = app
  end

  def call(env)
    ActiveSupport::Dependencies.clear
    @app.call(env)
  end
end

Jets.application.configure do
  config.middleware.use Reloader
end
```

This article is an interesting read: https://crypt.codemancers.com/posts/2013-10-03-rails-reloading-in-dev-mode/

## Drawbacks

It might be more complex 🤷‍♂ 

## Unresolved Questions

Maybe it's possible to use ActionDispatch::Reloader?

**Comment by tongueroo** [comment]
Yup, definitely want to improve the speed of development reloading.  Want to play with the reloader a more because it is more complex. 

As a workaround, some folks are have been using https://github.com/dblock/guard-rack  

Played with guard-rack also and noticed that it reports harmless errors if you save rapidly.  IE: Another reload happens before rack has fully restarted.

The errors are harmless but can be confusing for folks. May just roll out a guard-rack as an option for the reloader since it is so simple and quick to implement. Reloaders generally seem pretty complex, so think it may require some time to investigate 🤔

Happy to consider PRs here.

**Comment by dbackeus** [comment]
What about https://github.com/fxn/zeitwerk?

That's going to be the default code loader in Rails 6 and is supposedly designed to be used in any ruby project.

**Comment by tongueroo** [comment]
Yup. Would like to try zeitwerk also.

**Comment by tongueroo** [comment]
Addressed this in #254. Using zeitwerk for auto reloading in development mode. 

A quick check with `time curl localhost:8888/posts`

* Before: 0m2.259s
* After: 0m0.094s

24X improvement. Generally, improvement in the 10x range.

Even though a lot more can be done to clean up the loader, went ahead and rolled it out.

Released in [v1.9.0](https://github.com/tongueroo/jets/blob/master/CHANGELOG.md)
